gt/main.c

```diff
@@ -136,19 +137,30 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
 	if (inner_ip_ver == 4) {
 		info->inner_ip_ver = ETHER_TYPE_IPv4;
 		info->l4_proto = inner_ipv4_hdr->next_proto_id;
-		info->l4_hdr = &inner_ipv4_hdr[1];
+		info->l4_hdr = (void *)ipv4_skip_exthdr(inner_ipv4_hdr);
```

Comment:
I can't comment on the code above these lines, but we need to take care of variable network-layer headers while assigning variable `inner_ipv4_hdr` as well.